app/app/calculations/punching.py

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/env python
+# encoding: utf-8
+
+from math import pi, sin, cos, ceil
+from .consts import *
+
+ERRORS = []
+
+
+def compute_punching(c_class, s_class, dsw, support, section, b, h, dx, dy, lx,
+                     ly, ad, lambda_u, asx, asy, design_situation, ved, beta):
+    """
+    Procedura sprawdzenia nośności na przebicie płyty betonowej wg PN-EN 1992-1
+    """
+
+    # -----------------------------------------
+    # MATERIAL
+    # Beton
+    fck = c_tab[c_class.type] * MPa
+    # Stal
+    fyk = s_tab[s_class.type] * MPa
+    alfa_sw = 0.8           # redukcja wytrzymalosci zbr. na ścinanie, TODO(LZ): do namyslu jaka wartosc
+    
+    # -----------------------------------------
+    # GEOMETRIA
+    # Element = 'plyta'    # do wyboru: plyta, stopa fundamentowa TODO
+    cx = float(b) * cm     # cx = 40 * cm
+    cy = float(h) * cm     # cy = 40 * cm
+    #cxy = 40 * cm	       # TODO(LZ): dla przekroju kolowego
+    dsw = dsw.value * mm   # srednica zbrojenia na ścinanie
+    dx = float(dx) * cm
+    dy = float(dy) * cm
+    lx = float(lx) * cm    # odleglosc pola obciazenia do krawedzi plyty w kier. x
+    ly = float(ly) * cm    # jw. w kier. y
+    ad = float(ad) * cm
+    asx = float(asx) * cm2
+    asy = float(asy) * cm2    
+    lambda_u = lambda_u    # wsp. uwzgledniajacy otwor w poblizu obwodu kontrolnego (%)  # TODO: to przypisanie jest tu bez sensu, ale zrobiłam tak, żeby był ślad po lambda_u
```

Comment:
Tak naprawdę to można we wstępie opisać, co każdy parametr wejściowy znaczy i co otrzymujemy na wyniku. Jeśli skasujesz początkowy komentarz i wstawisz jeszcze raz """ to pycharm sam powinien wrzucić Ci strukturę, co można uzupełnić. Stworzyłaby się wtedy ładna dokumentacja. Jeśli piszę coś niejasno, to pytaj śmiało.